app/controllers/resources_controller.rb

```diff
@@ -59,20 +59,48 @@ def check_permission
     privilege = params[:privilege]
     raise ArgumentError, "privilege" unless privilege
 
-    result = assumed_role.allowed_to?(privilege, resource)
+    begin
+      result = assumed_role.allowed_to?(privilege, resource)
+
+      audit_success(privilege, result)
+    rescue => e
+      audit_failure(privilege, e)
+
+      raise
+    end
 
+    head(result ? :no_content : :not_found)
+  end
+
+  def audit_success(privilege, result)
     Audit.logger.log(
       Audit::Event::Check.new(
         user: current_user,
         client_ip: request.ip,
-        resource: resource,
+        resource_id: resource.id,
```

Comment:
So we are checking nil here yes?

Reply:
We are not checking nil because even before my changes to the file it was assumed that resource is not nil. I can add a check if you think thats necessary :)

Reply:
If resource was nil before this change we would have get:
`resource: nil` => will show nil at audit which is find, and no error is thrown

Now, after this change, we will get:
`resource: nil.id` => throwing error.

Reply:
It wouldnt just show nil at the audit because `check.new` calls methods of `check.rb` that use the id property of resource and role without any nil check. so it would throw an exception just from a different file